### PR TITLE
Quack

### PR DESCRIPTION
**Pull request by MadmanMartian**
A suggestion from Megautham, sprites provided by Megautham

It's a giant robot duck. Cute, cuddly, until you provoke it.

It will hunt you ceaselessly.

You can not run.

You can probably hide.

Fear the cute fluffy duck.

### CODE REVIEW

**Comment by jknpj** [comment]
That's horrifying

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Post sprites now that icondiffbot is kil

File: code/modules/mob/living/simple_animal/hostile/duck.dm [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+/mob/living/simple_animal/hostile/roboduck/OpenFire()
+	set waitfor = 0
+	playsound(src, 'sound/misc/quacktivated.ogg', 40, 5, 4)
+	do_flick(src, "[initial(icon_state)]_gunfetti_start", 5)
+	icon_state = "[initial(icon_state)]_gunfetti_loop"
+	canmove = 0
+	var/volleys = rand(2,5)
+	for(var/i = 0,i < volleys, i++)
+		for(var/direction in alldirs)
+			if(gcDestroyed)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/duck.dm]: I think you should check this after the sleep, not before

File: code/modules/mob/living/simple_animal/hostile/duck.dm [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+/mob/living/simple_animal/hostile/roboduck/update_icon()
+	if(angered)
+		icon_state = "[initial(icon_state)]_hostile"
+	else
+		icon_state = "[initial(icon_state)]_friendly"
+
+/mob/living/simple_animal/hostile/roboduck/Aggro()
+	..()
+	if(!angered && do_flick(src, "[initial(icon_state)]_angered"))
+		playsound(src, 'sound/misc/quacktivated.ogg', 40, 5)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/duck.dm]: vary=5? what did you mean by this?

File: code/modules/mob/living/simple_animal/hostile/duck.dm [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		angered = TRUE
+		update_icon()
+
+/mob/living/simple_animal/hostile/roboduck/LoseAggro()
+	..()
+	if(angered && do_flick(src, "[initial(icon_state)]_calming"))
+		playsound(src, 'sound/misc/roboquack.ogg', 40, 5)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/hostile/duck.dm]: vary=5? what did you mean by this?

**Pull request merged**